feat(config): Switch work order files from JSON to YAML

Ishan1923 opened the pull request:
Closes #86

This pull request migrates the work order description files from JSON to YAML format, as proposed in issue #86.

Ishan1923 commented:
Hi @aaronchongth,

Apologies for the delay in getting these updates to you, and thanks again for the detailed review.

I have now pushed the latest changes addressing all your feedback. Here is a quick summary of the work done:

* The test scripts in `nexus_demos` have been updated to use and parse the new `.yaml` work order files.
* `python3-yaml` has been added as a dependency to the `nexus_demos` package.xml.
* The EOF nitpicks in the three `.yaml` config files have been corrected.
* The filename reference in the disabled test file (`test_parallel_pick_and_place_rmf.py.disabled`) has been fixed.

The CI checks should all be passing now. Could you please take another look when you have a moment?

Best,
Ishan

aaronchongth commented on nexus_demos/package.xml:
```diff
@@ -67,6 +67,9 @@
   <test_depend>ament_index_cpp</test_depend>
   <test_depend>rcpputils</test_depend>
 
+  <!-- json2yaml edit -->
```
Remove stale comment

aaronchongth reviewed (changes requested):
Thanks for the changes, I have left a few more comments.

We have not made the necessary changes to the github repository template yet, but could you also declare in your PR description if Generative AI tools were used for this PR, per OSRA policy https://discourse.openrobotics.org/t/osrf-adopts-policy-on-use-of-generative-ai-in-contributions/43685, and if so, what tool.

You can add and fill up this block
```
### GenAI Use
We follow [OSRA's policy on GenAI tools](https://github.com/openrobotics/osrf-policies-and-procedures/blob/main/OSRF%20Policy%20on%20the%20Use%20of%20Generative%20Tools%20(%E2%80%9CGenerative%20AI%E2%80%9D)%20in%20Contributions.md)

- [ ] I used a GenAI tool in this PR.
- [ ] I did not use GenAI

Generated-by:
```

Ishan1923 commented:
Hi @aaronchongth,

I've just pushed the latest updates addressing your feedback. The stale comments and extra whitespace have been removed from the YAML files, and I've also added the GenAI declaration to the pull request description.

Could you please take another look when you have a moment?

Thanks!
